controllers/ovndbcluster_controller.go

```diff
@@ -581,6 +590,76 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			}
 		}
 	}
+
+	Log.Info("DNS Starts here")
+	if instance.Spec.DBType == v1beta1.SBDBType {
+		Log.Info("Only doing it for SB pods")
+		serviceName = ovndbcluster.ServiceNameSB
+		serviceLabels = map[string]string{
+			common.AppSelector: serviceName,
+		}
+		podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		for _, ovnPod := range podList.Items {
+			Log.Info(fmt.Sprintf("Using Pod: %v", ovnPod.Name))
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = "dns-" + ovnPod.Name
+			// Get Hostname
+			svc, _ := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			hostname := svc.ObjectMeta.Annotations["dnsmasq.network.openstack.org/hostname"]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			for _, v := range net_stat {
+				if v.Name == "openstack/internalapi" {
+					dnsIP = v.IPs[0]
+				}
+			}
+
+			// Create DNSRecord
+			var DNSRecords []infranetworkv1.DNSHost
+			DNSRecord := infranetworkv1.DNSHost{}
+			DNSRecord.IP = dnsIP
+			DNSRecord.Hostnames = hostnames
+			DNSRecordCname := infranetworkv1.DNSHost{}
+			DNSRecordCname.IP = dnsIP
+			DNSRecordCname.Hostnames = append(DNSRecordCname.Hostnames, "ovsdbserver-sb.openstack.svc")
+			DNSRecords = append(DNSRecords, DNSRecord)
+			DNSRecords = append(DNSRecords, DNSRecordCname)
+
+			// Create DNSData
+			DNSData := &infranetworkv1.DNSData{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dnsName,
+					Namespace: ovnPod.Namespace,
+				},
+			}
+			_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), DNSData, func() error {
+				DNSData.Spec.Hosts = DNSRecords
+				DNSData.Spec.DNSDataLabelSelectorValue = "dnsdata"
```

Comment:
this defaults to `dnsdata` so no need to set it, but it won't hurt

Reply:
I see "dnsdata" defaults is with kubebuilder so it won't work here when we create the instances with golang structs. @stuggi is this right or i miss here something?
Also would be better to reuse the value from DNSMasq instance instead of hardcoding it here

Reply:
Update, since it was the default I removed it, it turns out that the DNSData will be created with `DNSDataLabelSelectorValue` set to "", making it impossible to resolve the host from the edpm compute.
```
spec:
  dnsDataLabelSelectorValue: ""
  hosts:
  - hostnames:
    - ovsdbserver-nb-0.openstack.svc
    ip: 172.17.0.31
```
Reseting the value.

Reply:
@karelyatin you are right, sorry for the confusion.